### ISSUE TITLE
All the rows of a table are dropped automatically

### ISSUE THREAD

**Issue by torresl** [balderdashy/sails#2877]
Hi,

  I created a table with a model  but after a certain time  all  data of this table  are  dropped doing nothing .
my models.js file :   
"  
module.exports.models = {
 
  // Your app's default connection.
  // i.e. the name of one of your app's connections (see `config/connections.js`)
  //
  // (defaults to localDiskDb)
  connection: 'memory',
  migrate: 'alter'
};
"

What must i do to not have data dropped automatically ?.

Thanks

**Comment by tjwebb** [comment]
You're using the memory adapter. You should use a real database if you want your data to stay around.

**Comment by keniosor** [comment]
No,  I don't use 'memory' for this table wich is dropped. I use an adapter mysql in the connection.js file :     
 mysqljj: {
  	
	    adapter: 'sails-mysql',
	    host: 'localhost',
	    port: 3306,
	    database: 'xxxx',	
	    user: 'xxxxx',
	    password: 'xxx'
	    
	  },     

Perhaps must i comment  " connection: 'memory', " ?

**Comment by tjwebb** [comment]
Your connection is set to memory: 
> connection: 'memory',

If you want to use that mysql connection, set `connection: 'mysqljj`

**Comment by keniosor** [comment]
I associate the connection 'mysqljj' directly in the model file of the table.  So, i don' t use the default connection in the models.js file.

**Comment by tjwebb** [comment]
Post a link to your project and specify which models/rows are being dropped.

**Comment by lvsenlin** [comment]
models.js set like this:
  migrate: 'safe',
  //migrate: 'alter',
  //migrate: 'drop',
dont's use 'alter' or 'drop',it will delete you data if you modify your models.

**Comment by sahanDissanayake** [comment]
http://stackoverflow.com/questions/29997983/best-way-to-migrate-table-changes-to-production-sailsjs-tables/29998429#29998429

https://github.com/balderdashy/sails/issues/2901

**Comment by tcury** [comment]
I had this problem today. I have 'migrate: alter' and my table user was truncated.
+1 to fix it!

**Comment by sailsbot** [comment]
Thanks for posting, @torresl.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
